### PR TITLE
Test for question control servlets

### PR DESCRIPTION
**Pull request by pamelalozano16**
PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the new behavior?

For now it's just the StartGameInstanceServlet test

### CODE REVIEW

**Comment by pamelalozano16** [comment]
Guys I don't want to merge this branch yet, but I'm doing this pull request so you can review this test I did, I'd like all of your reviews because it's the first one I do and I want to make sure I'm going in the right path. Thank you u are awesome :)

**Comment by pamelalozano16** [comment]
> Guys I don't want to merge this branch yet, but I'm doing this pull request so you can review this test I did, I'd like all of your reviews because it's the first one I do and I want to make sure I'm going in the right path. Thank you u are awesome :)

While creating the test I thought of refactoring the StartGameInstanceServlet by using GameDao to get game, that way I wouldn't need to initialize firestore in the test, but we don't have a getGame in GameDao yet.

File: src/test/java/com/google/sps/StartGameInstanceServletTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          .questions(new ArrayList<Question>())
+          .build();
+
+    when(mockGameDao.getGame(gameId)).thenReturn(newGame);
+
+    //Simulate Room to be updated
+    GameInstance newRoom = new GameInstance(roomId);
+    newRoom.setGameId(gameId);
+    newRoom.setCreator("UfQ5TlrtFtNJnR1ywXb8W7Hrz6y1");
+    newRoom.setIsActive(true);
```
Comment (brettapeters) [inline src/test/java/com/google/sps/StartGameInstanceServletTest.java:99]: Start out with "isActive" and "currentQuestion" unset before calling `doPost`. Then you can verify their values before/after the servlet code sets them. You could capture the `GameInstance` argument from the call to `updateGameInstance` and assert that its "isActive" and "currentQuestion" fields are certain values.

https://www.javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/ArgumentCaptor.html

**Comment by brettapeters** [comment]
> > Guys I don't want to merge this branch yet, but I'm doing this pull request so you can review this test I did, I'd like all of your reviews because it's the first one I do and I want to make sure I'm going in the right path. Thank you u are awesome :)
> 
> While creating the test I thought of refactoring the StartGameInstanceServlet by using GameDao to get game, that way I wouldn't need to initialize firestore in the test and it would be much faster, but we don't have a getGame in GameDao yet.

Tests shouldn't import anything Firestore related. All you need are the DAO interfaces to use for mocks.

File: src/test/java/com/google/sps/StartGameInstanceServletTest.java (hunk 1 above)
Comment (pamelalozano16) [reply src/test/java/com/google/sps/StartGameInstanceServletTest.java:99]: Done! Thanks brett

**Pull request merged**